### DISCUSSION TITLE
[Question]: How to handle objects half out of frame when verifying in frigate+ ?

### DISCUSSION THREAD

**Discussion by JohannCR** [blakeblackshear/frigate#11346] [Ask A Question] [answered]
### What is your question:

Sometimes, an object is detected while half out of frame. Is it ok to label it anyway or should we only use images where the objects are not cut off ?
For example a car sitting in the corner of a frame, visible at around 60 to 80% can be included and labeled ?
I know that if part of an object is hidden we need to tag its entirety, including the non visible parts to have a correct ratio, but this is not possible when the object is partly out of the frame.

**Comment by JohannCR** [comment]
Also the paid upgrade is "coming soon", does it mean it's not available yet, or maybe not in every country ?
That's too bad, I just spent a couple hours annotating a bunch  of images thinking I would try it out 🤣

**Reply by hawkeye217** [reply]
Just email blake@frigate.video

**Reply by JohannCR** [reply]
Thanks, that settles this !
Only question left is how to handle cut objects. Any suggestion ?

**Reply by hawkeye217** [reply, marked as answer]
If a car is between 60-80% visible, yes, go ahead and label it.

If you haven't seen the docs on labeling and improving your model check them out here: https://docs.frigate.video/plus/improving_model

**Reply by JohannCR** [reply]
I did ;-) thanks for the confirmation on the cut objects.
Maybe we should add that to the documentation, people are bound to encounter some situations that raise questions (like if a snapshot labels an object in one place but the object has moved to a different place in the meantime, or when labeling faces is there a minimum size/pixel definition, do we label the whole head or just the face, stuff like that).
I know data quality is paramount for the training dataset (I work in the field^^), so since it's a shared dataset, documentation is important unless there are safeguards to detect wrong labels...

**Reply by NickM-27** [reply]
Not every image that a user submits is added to the base model that is used for other user's. Blake picks them by hand and validates / fixes those images. So submitting incorrect images would only affect a users own model. The process was outlined in detail at https://github.com/blakeblackshear/frigate/discussions/9466

**Reply by JohannCR** [reply]
Wow that thread is very interesting, full of nuggets ! Thx
FYI made a pull request to update the docs

**Comment by JohannCR** [comment]
Hi,
I'd like to insist on one question I mentioned in a comment (in this thread) that hasn't been answered yet (and I haven't found any info on this elsewhere)^^
Asking again with more details :
- Is it better to ALWAYS tag faces when we tag a person (seen from the front of course) ? Even if the person is far away in the image so the face is just a blur of pixels ?
- Also does the face mean just the face features (eyes, nose, mouth) or should we include ears, hair and chin ?
- Finally, should we tag faces seen from the side ?

It may seem like dumb questions, but depending on what the model is trained to recognized as a face, consistency is kind of important...
Thanks !

**Reply by NickM-27** [reply]
> Is it better to ALWAYS tag faces when we tag a person (seen from the front of course) ? Even if the person is far away in the image so the face is just a blur of pixels ?

if you can identify that it is a face then yes, but if the face is only a few pixels then it probably doesn't make sense

> Also does the face mean just the face features (eyes, nose, mouth) or should we include ears, hair and chin ?

I typically just label the eyes, nose, mouth

> Finally, should we tag faces seen from the side ?

if you can tell that it is a face (I usually try to see if the nose is visible)
